src/main/java/org/opentripplanner/netex/mapping/StopTimesMapper.java

```diff
@@ -168,17 +168,23 @@ private StopTime mapToStopTime(
         stopTime.setStop(stop);
         if (passingTime.getArrivalTime() != null || passingTime.getDepartureTime() != null) {
             stopTime.setArrivalTime(calculateOtpTime(
-                passingTime.getArrivalTime(),                 
-                passingTime.getArrivalDayOffset(),
-                passingTime.getDepartureTime(),
-                passingTime.getDepartureDayOffset()
+                    passingTime.getArrivalTime(),
+                    passingTime.getArrivalDayOffset(),
+                    passingTime.getDepartureTime(),
+                    passingTime.getDepartureDayOffset()
             ));
             stopTime.setDepartureTime(calculateOtpTime(
-                passingTime.getDepartureTime(),
-                passingTime.getDepartureDayOffset(),
-                passingTime.getArrivalTime(),
-                passingTime.getArrivalDayOffset()
+                    passingTime.getDepartureTime(),
+                    passingTime.getDepartureDayOffset(),
+                    passingTime.getArrivalTime(),
+                    passingTime.getArrivalDayOffset()
             ));
+
+            // From NeTEx we define timepoint as a waitpoint with waiting time defined (also 0)
+            if (Boolean.TRUE.equals(stopPoint.isIsWaitPoint())
```

Comment:
This looks slightly ugly, but is required as the value can be null. Should we do a null check and equals to the primitive type instead?

Reply:
I disagree, it uses well documented SDK objects and functions.

Reply:
I kinda agree with @TimMickelsonSkanetrafiken. Personally, I find the null checks that Java code bases are full of ugly to look at.